pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1221,6 +1211,23 @@ func Predicates(pod *v1.Pod, nodeInfo *schedulercache.NodeInfo) (bool, []algorit
 	return len(predicateFails) == 0, predicateFails, nil
 }
 
+func NodeConditionPredicates(nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason) {
+	reasons := []algorithm.PredicateFailureReason{}
+
+	for _, c := range nodeInfo.Node().Status.Conditions {
+		// TODO: There are other node status that the DaemonSet should ideally respect too,
+		//       e.g. MemoryPressure, and DiskPressure
+		if c.Type == v1.NodeOutOfDisk && c.Status == v1.ConditionTrue {
```

Comment:
@gmarek @janetkuo , should we add other node status here? or tolerate other status in #42001 ?

And I'm also thinking should we taint 'MemoryPressure' and 'DiskPressure' in #42001 ? Should we ask kubelet to help on it? and in scheduler, we only check NodeReady, NetworkUnavailable and OutOfDisk condition.